### DISCUSSION TITLE
Can I take repository from Github GraphQl Api by ID

### DISCUSSION THREAD

**Discussion by hajczek** [community/community#22184] [New to GitHub]
<p>Hi,</p>
<p>I want to take details about the repository by using repository ID.<br>
Is it possible?<br>
If yes, how to do that?<br>
I don’t know, in which way I must prepare the query for that.</p>
<p>Thanks for help.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/hajczek">@hajczek</a>, welcome to the community! Yes, you can absolutely do that, try the query below. Here’s an example using this repository:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/AndreaGriffiths11/graphql-test" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars1.githubusercontent.com/u/20666190?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/AndreaGriffiths11/graphql-test" target="_blank" rel="noopener">AndreaGriffiths11/graphql-test</a></h3>


  <p><span class="label1">a repository with test queries, to test queries. Contribute to AndreaGriffiths11/graphql-test development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The GitHub API exposes the <code>node_id</code> when fetching a repository:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <a href="https://api.github.com/repos/AndreaGriffiths11/graphql-test" target="_blank" rel="noopener">api.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://api.github.com/repos/AndreaGriffiths11/graphql-test" target="_blank" rel="noopener"></a></h3>

<p><a href="https://api.github.com/repos/AndreaGriffiths11/graphql-test" target="_blank" rel="noopener">//api.github.com/repos/AndreaGriffiths11/graphql-test</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>In this case, the <code>node_id</code> is <code>MDEwOlJlcG9zaXRvcnkxODEyMDQwMDI</code> .</p>
<p>Using our <a href="https://docs.github.com/en/free-pro-team@latest/graphql/overview/explorer">GraphQL API Explorer</a>, you can run this query to fetch that repository by its <code>node_id</code> :</p>
<pre><code class="lang-auto">query { 
    node(id: "MDEwOlJlcG9zaXRvcnkxODEyMDQwMDI"){
    ... on Repository {
      id
      nameWithOwner
    }
  }
}
</code></pre>
<p>I hope this helps!<br>
AG</p>

**Comment by hajczek** [comment]
<p>Hi <a class="mention" href="/AndreaGriffiths11">@AndreaGriffiths11</a>,</p>
<p>Thank you so much for this query, is what I want but not exactly.</p>
<p>I have the app where I want to search for details about the repository using the repository id, but dynamically declared. It means that I have this ‘id’ in some variable and I want to put this variable inside query, something like that:</p>
<pre><code class="lang-auto">query { 
    node(id: $id){
    ... on Repository {
      id
      nameWithOwner
    }
  }
}
</code></pre>
<p>with declared variable $id.</p>
<p>But when I try to do this in Explorer, I have this message:  <strong>"Variable $id is used by  but not declared</strong>"</p>
<p>How can I do this, maybe you know?</p>
<p>Thanks for help.</p>

**Comment by hajczek** [comment, marked as answer]
<p>Thanks for help again!</p>
<p>I found the solution of my problem :slight_smile:</p>
<p>Here is my query, which works well:</p>
<pre><code class="lang-auto">query repository($id:  ID!){ 
    node(id: $id){
    ... on Repository {
      id
      nameWithOwner
    }
  }
} 
</code></pre>
<p>Have a nice evening!!!</p>

**Reply by luckydonald** [reply]
Is that faster or slower than using `owner` + `name`?

**Comment by AndreaGriffiths11** [comment]
<p>Awesome :clap: thanks for coming back to share your solution.</p>
